src/app/modules/form/admin-form/admin-form.controller.ts

```diff
@@ -1494,3 +1494,50 @@ export const handleUpdateFormField = [
   ),
   _handleUpdateFormField,
 ]
+/*
+ * Handler for DELETE /forms/:formId/logic/:logicId
+ * @security session
+ *
+ * @returns 200 with success message when successfully deleted
+ * @returns 403 when user does not have permissions to delete logic
+ * @returns 404 when form cannot be found
+ * @returns 422 when user in session cannot be retrieved from the database
+ * @returns 500 when database error occurs
+ */
+export const handleDeleteLogic: RequestHandler = (req, res) => {
+  const { formId, logicId } = req.params
+  const sessionUserId = (req.session as Express.AuthedSession).user._id
+
+  // Step 1: Retrieve currently logged in user.
+  return (
+    UserService.getPopulatedUserById(sessionUserId)
+      .andThen((user) =>
+        // Step 2: Retrieve form with write permission check.
+        AuthService.getFormAfterPermissionChecks({
+          user,
+          formId,
+          level: PermissionLevel.Write,
+        }),
+      )
+      // Step 3: Delete form logic
+      .andThen((retrievedForm) =>
+        AdminFormService.deleteFormLogic(retrievedForm, logicId),
+      )
+      .map(() => res.json({ message: 'Logic deleted successfully' }))
```

Comment:
rather than sending a superfluous message which we don't intend for the client to use, might it be better to just do `res.sendStatus(200)`?

Reply:
edited